ビルドエラー

```
src/endpoints/talks/messages/delete.ts(39,4): error TS2349: Cannot invoke an expression whose type lacks a call signature. Type '((fn?: (err: any, product: ITalkUserMessage, numAffected: number) => void) => Promise<ITalkUserMe...' has no compatible call signatures.
```

--- a/src/endpoints/talks/messages/delete.ts
+++ b/src/endpoints/talks/messages/delete.ts
@@ -14,7 +14,7 @@ export default function(user: IUser, messageId: string): Promise<Object> {
 
 	return new Promise<Object>((resolve, reject) => {
 		// 対象のメッセージを取得
-		TalkMessage.findOne({_id: messageId, user: user.id}, (findErr: any, message: ITalkUserMessage | ITalkGroupMessage) => {
+		TalkMessage.findOne({_id: messageId, user: user.id}, (findErr: any, message: any) => {
 			if (findErr !== null) {
 				return reject(findErr);
 			} else if (message === null) {

--- a/src/endpoints/talks/messages/delete.ts
+++ b/src/endpoints/talks/messages/delete.ts
@@ -2,6 +2,10 @@ import {TalkMessage} from '../../../db/db';
 import {ITalkMessage, ITalkUserMessage, ITalkGroupMessage, IUser} from '../../../db/interfaces';
 import event from '../../../event';
 
+function isUserMessage(message: ITalkUserMessage | ITalkGroupMessage): message is ITalkUserMessage {
+	return message.type === 'user-message';
+}
+
 /**
  * Talkのメッセージを削除します
  * @param user API利用ユーザー
@@ -32,7 +36,7 @@ export default function(user: IUser, messageId: string): Promise<Object> {
 				} else {
 					resolve();
 
-					if (message.type === 'user-message') {
+					if (isUserMessage(message)) {
 						event.publishDeleteTalkUserMessage(user.id, <string>message.recipient, message);
 					} else {
 						event.publishDeleteTalkGroupMessage(<string>message.group, message);

--- a/src/endpoints/talks/messages/delete.ts
+++ b/src/endpoints/talks/messages/delete.ts
@@ -2,29 +2,19 @@ import {TalkMessage} from '../../../db/db';
 import {ITalkMessage, ITalkUserMessage, ITalkGroupMessage, IUser} from '../../../db/interfaces';
 import event from '../../../event';
 
-function isUserMessage(message: ITalkMessage): message is ITalkUserMessage {
-	return message.type === 'user-message';
-}
-
 /**
  * Talkのメッセージを削除します
  * @param user API利用ユーザー
  * @param messageId メッセージのID
  */
-export default function(
-	user: IUser,
-	messageId: string
-): Promise<void> {
+export default function(user: IUser, messageId: string): Promise<Object> {
 	if (messageId === '')  {
-		return <Promise<any>>Promise.reject('empty-message-id');
+		return <Promise<Object>>Promise.reject('empty-message-id');
 	}
 
-	return new Promise<void>((resolve, reject) => {
+	return new Promise<Object>((resolve, reject) => {
 		// 対象のメッセージを取得
-		TalkMessage.findOne({
-			_id: messageId,
-			user: user.id
-		}, (findErr: any, message: ITalkUserMessage | ITalkGroupMessage) => {
+		TalkMessage.findOne({_id: messageId, user: user.id}, (findErr: any, message: ITalkUserMessage | ITalkGroupMessage) => {
 			if (findErr !== null) {
 				return reject(findErr);
 			} else if (message === null) {
@@ -38,15 +28,15 @@ export default function(
 			message.isDeleted = true;
 			message.save((saveErr: any) => {
 				if (saveErr !== null) {
-					return reject(saveErr);
-				}
-
-				resolve();
-
-				if (isUserMessage(message)) {
-					event.publishDeleteTalkUserMessage(user.id, <string>message.recipient, message);
+					reject(saveErr);
 				} else {
-					event.publishDeleteTalkGroupMessage(<string>message.group, message);
+					resolve();
+
+					if (message.type === 'user-message') {
+						event.publishDeleteTalkUserMessage(user.id, <string>message.recipient, message);
+					} else {
+						event.publishDeleteTalkGroupMessage(<string>message.group, message);
+					}
 				}
 			});
 		});

